.github/workflows/pull_request.yml

```diff
@@ -1,5 +1,31 @@
 name : Reviewdog PR Annotations
-on: [pull_request_target]
+
+# =========
+# IMPORTANT
+# =========
+#
+# TL;DR:
+#     THIS ACTION SHOULD NOT RUN ANY CODE FROM THE PR BRANCH.
+#
+# This action is triggered after new events in Pull Requests (as the
+# `pull_request` trigger does), but this ones provides the workflow writing
+# permissions to the repo.
```

Comment:
I would add a note clarifying that the version of this workflow file that is actually executed is the one on the pull request's TARGET branch, not the one on pull request's SOURCE. 

It's a subtle, but important difference, because it runs from the target branch (usually main), thus why it gets elevated permissions,

Reply:
We should probably include a link to the original pr this was added in for further reference.